src/Control/Distributed/Process/Internal/Types.hs

```diff
@@ -366,6 +373,10 @@ data Message =
   }
   deriving (Typeable)
 
+instance NFData Message where
+  rnf (EncodedMessage _ e) = rnf e `seq` ()
+  rnf (UnencodedMessage _ a) = a `seq` ()   -- forced to WHNF only
```

Comment:
In unencoded message we have existential variable without NFData constraint, so we just can't force it to NF. This semantics was not changed in this pull request. AFAIU the reason for this change is ability to pass messages locally without additional serialization or forcing.